src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -478,6 +478,11 @@ namespace Microsoft.FSharp.Core
         let inline isinstPrim<'T>(x:obj) = (# "isinst !0" type ('T) x : obj #)
         let inline castclassPrim<'T>(x:obj) = (# "castclass !0" type ('T) x : 'T #)
         let inline notnullPrim<'T when 'T : not struct>(x:'T) = (# "ldnull cgt.un" x : bool #)
+           
+        let inline internal notnullFast (x: ^T) = 
```

Comment:
This might be a useful function to have outside of F# core. Why not expose it as public?